### PR TITLE
Enabled multi-viewport rendering

### PR DESCRIPTION
**Pull request by peterAdams1014**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -2594,201 +2595,255 @@ void vcMain_RenderSceneWindow(vcState *pProgramState)
...
+      ImGui::Image(renderData.pSceneTexture, ImVec2((float)pProgramState->pActiveViewport->resolution.x, (float)pProgramState->pActiveViewport->resolution.y), ImVec2(0, 0), ImVec2(renderData.sceneScaling.x, renderData.sceneScaling.y));
+
+      // TODO: Screenshot which viewport?
+      if (viewportIndex == 0 && pProgramState->settings.screenshot.taking)
+        pProgramState->screenshot.pImage = renderData.pSceneTexture;
+
+      static bool wasContextMenuOpenLastFrame = false;
+      bool useTool = (io.MouseDragMaxDistanceSqr[0] < (io.MouseDragThreshold * io.MouseDragThreshold)) && ImGui::IsMouseReleased(0) && ImGui::IsItemHovered();
+
+      // TODO: TODO: global state in here makes it not work for multiple viewports
```
Comment (samsinsane) [inline src/main.cpp:2651]: Is this a TODO to do a TODO?

File: src/main.cpp [hunk 2]
```diff
@@ -2594,201 +2595,255 @@ void vcMain_RenderSceneWindow(vcState *pProgramState)
...
 
-    if ((io.MouseDragMaxDistanceSqr[1] < (io.MouseDragThreshold * io.MouseDragThreshold) && ImGui::BeginPopupContextItem("SceneContext")))
-    {
-      static bool hadMouse = false;
-      static udDouble3 mousePosCartesian;
-      static udDouble3 mousePosLongLat;
+      pProgramState->pActiveViewport = &pProgramState->pViewports[viewportIndex];
 
-      if (!wasContextMenuOpenLastFrame || ImGui::IsMouseClicked(1))
+      // TODO: Screenshot which viewport?
```
Comment (samsinsane) [inline src/main.cpp:2628]: Both as a single image - if the user wants a screenshot of just one, they should close the other viewport.

File: src/main.cpp (hunk 1 above)
Comment (peterAdams1014) [reply src/main.cpp:2651]: :P

File: src/main.cpp (hunk 2 above)
Comment (peterAdams1014) [reply src/main.cpp:2628]: Sounds reasonable.

**Pull request merged**